src/containers/blocks.jsx

```diff
@@ -172,6 +172,11 @@ class Blocks extends React.Component {
         this.workspace.reportValue(data.id, data.value);
     }
     onWorkspaceUpdate (data) {
+        // When we change sprites, update the toolbox to have the new sprite's blocks
+        if (this.props.vm.editingTarget) {
+            this.props.updateToolboxState(makeToolboxXML(this.props.vm.editingTarget.id));
+        }
```

Comment:
I'm ok with doing this until we figure out a better solution, but this is going to cause `updateToolbox` to be called twice: once as a result of this, and once as a result of the workspace update below. We should be able to keep that down to once, although it might involve some blocks change.

Reply:
I've added a todo and filed issue https://github.com/LLK/scratch-gui/issues/901